datasets/librispeech.py

```diff
@@ -0,0 +1,180 @@
+        assert b"OK" in checksum_result
+
+        subprocess.check_call(["tar", "-xf", "%s.tar.gz" % self.corpus_key, "-C", tk.uncached_path(self._out_parent_folder)])
+
+
+class DownloadLibriSpeechMetadataJob(DownloadLibriSpeechCorpusJob):
+    """
+    Downloads the metadata file and checks for md5sum integrity
+
+    Defines outputs for "SPEAKERS.TXT, CHAPTERS.TXT and BOOKS.TXT"
+    """
+
+    # noinspection PyMissingConstructor
+    def __init__(self):
+        self.corpus_key = "raw-metadata"
+
+        # this is just needed to have the extraction target available
+        self._out_parent_folder = self.output_path(".", directory=True)
+
+        self.out_speakers = self.output_path("LibriSpeech/SPEAKERS.TXT")
+        self.out_chapters = self.output_path("LibriSpeech/CHAPTERS.TXT")
+        self.out_books = self.output_path("LibriSpeech/BOOKS.TXT")
+
+
+class LibriSpeechCreateBlissJob(Job):
+    """
+    Creates a Bliss corpus from a LibriSpeech corpus folder using the speaker information in addition
+
+    Outputs a single bliss .xml.gz file
+    """
+
+    def __init__(self, corpus_folder, speaker_metadata):
+        """
+        Generate a bliss xml from  the LibriSpeech corpus.
+
+        :param Path corpus_folder: Path to a LibriSpeech corpus folder
+        :param Path speaker_metadata: Path to SPEAKER.TXT file from the MetdataJob (out_speakers)
+        """
+        self.corpus_folder = corpus_folder
+        self.speaker_metadata = speaker_metadata
+
+        self.out_corpus = self.output_path("corpus.xml.gz")
+
+        self._speakers = {}  # dict(key: id, value: [sex, subset, min, name]
+        self._transcripts = []  # [dict(name, chapter, segment, orth, path)]
+
+    def tasks(self):
+        yield Task('run', mini_task=True)
+
+    def run(self):
+        self.get_speakers()
+        self.get_transcript()
+
+        c = corpus.Corpus()
+        c.name = os.path.basename(self.corpus_folder)
+
+        for speaker_id, speaker_info in sorted(self._speakers.items()):
+            speaker = corpus.Speaker()
+            speaker.name = speaker_id
+            speaker.attribs['gender'] = speaker_info[0].lower()
+            c.add_speaker(speaker)
+
+        for transcript in self._transcripts:
+            name = "{0}-{1}-{2:04d}".format(
+                transcript['speaker_id'],
+                transcript['chapter'],
+                transcript['segment']
+            )
+            recording = corpus.Recording()
+            recording.name = name
+            recording.speaker_name = transcript['speaker_id']
+            recording.audio = "{}/{}.flac".format(transcript['path'], name)
+
+            segment = corpus.Segment()
+            segment.name = name
+            segment.start = 0
+            segment.end = "inf"
+            segment.orth = transcript['orth'].strip()
+
+            recording.segments.append(segment)
+            c.recordings.append(recording)
+
+        c.dump(tk.uncached_path(self.out_corpus))
```

Comment:
is there any difference between tk.uncached_path(self.out_corpus) and self.out_corpus.get_path()? do we have a convention for this? since out_ vars schould be Path objects and not str

Reply:
I think so far not, but then we should decide on one.

Reply:
@curufinwe your take?

Reply:
For the out_ paths I'd go for `.get_path()` since we know it's a path obj and `.get_path` is a bit shorter.